packages/kit/src/runtime/server/page/server_routing.js

```diff
@@ -0,0 +1,81 @@
+import { s } from '../../../utils/misc.js';
+import { has_server_load } from '../../../utils/routing.js';
+import { load_page_and_error_nodes } from './load_page_nodes.js';
+import { text } from '../../../exports/index.js';
+
+/**
+ * @param {URL} url
+ * @param {import('types').SSROptions} options
+ * @returns {boolean}
+ */
+export function is_route_resolution_request(url, options) {
+	return (
+		url.pathname === `/${options.app_dir}/routes.js` ||
+		url.pathname.startsWith(`/${options.app_dir}/routes/`)
```

Comment:
I suspect all these functions will need to become basepath-aware (or their callers will)